src/main/java/org/testng/xml/internal/XmlSuiteUtils.java

```diff
@@ -81,6 +70,35 @@ public static XmlSuite newXmlSuiteUsing(List<String> classes) {
         xmlTest.setXmlClasses(constructXmlClassesUsing(classes));
         return xmlSuite;
     }
+    
+    public List<XmlSuite> getCloneSuite() {
+        return cloneSuites;
+    }
+
+    /**
+     * @param testNames input from m_testNames
+     * 
+     */
+    public List<String> getMissMatchedTestNames(List<String> testNames){
```

Comment:
Why not:

```java
public List<String> getMissMatchedTestNames(List<String> testNames){
        List<String> tmpTestNames = Lists.newArrayList();
        for (String testName : testNames) {
            if (matchedTestNames.contains(testName)) {
                tmpTestNames.add(testName);
            }
        }
        return tmpTestNames;        
    }
```

Reply:
No... I don't think tmpTestNames in your example can return miss-matched test names.